mne/evoked.py

```diff
@@ -1436,6 +1436,73 @@ def _write_evokeds(fname, evoked, check=True):
         end_file(fid)
 
 
+@fill_doc
+def export_evokeds(fname, evoked, fmt='auto', *, mff_device=None,
+                   mff_history=None):
+    """Export evoked dataset to external formats.
+
+    Supported formats: MFF (mff, uses module mffpy)
+    %(export_warning)s :func:`mne.write_evokeds` instead.
+
+    Parameters
+    ----------
+    %(export_params_fname)s
+    evoked : Evoked instance, or list of Evoked instances
+        The evoked dataset, or list of evoked datasets, to export to one file.
+        Note that the measurement info from the first evoked instance is used,
+        so be sure that information matches.
+    fmt : 'auto' | 'mff'
+        Format of the export. Defaults to ``'auto'``, which will infer the
+        format from the filename extension. See supported formats above for
+        more information.
+    mff_device : None (default) | str
+        If exporting to MFF format, it is required to specify the device on
+        which EEG was recorded (e.g. 'HydroCel GSN 256 1.0'). This is necessary
+        for determining the sensor layout and coordinates specs.
+    mff_history : None (default) | list of dict
+        If exporting to MFF format, it is optional to provide a list of history
+        entries (dictionaries) to be written to history.xml. This must adhere
+        to the format described in mffpy.xml_files.History.content. If None, no
+        history.xml will be written.
```

Comment:
I am a bit uneasy about this. Ideally the export_evokeds function should have a format agnostic signature. Otherwise we have one export function per format. Also I see the word "required" while it defaults to None. I assume it would be highly recommended to save this information. Can we infer the device from channel names and/or their locations?

Reply:
> Otherwise we have one export function per format.

@agramfort I could live with this, as to me it's mostly the same. However, it goes against how things work for Raw and Epochs currently, so we'd want to switch those to have separate functions, too, otherwise the inconsistent interface isn't worth it to me. And this would require a deprecation cycle because we already released with the EEGLAB export.

Reply:
Maybe it makes more sense to just have `mne.io.egi.export_evokeds_to_mff` as the API for exporting to MFF.

The device argument determines which `sensorLayout.xml` and `coordinates.xml` to write to the MFF directory. These are generic files that are determined only by the net type used to record EEG and they give the 2D and 3D positions of the sensors (see https://github.com/BEL-Public/mffpy/tree/develop/mffpy/resources/coordinates). The tricky part is that sometimes we have two different net types (e.g. 'HydroCel GSN 32 1.0' and 'MicroCel GSN 100 32 1.0') with the exact same sensor positions.

I think it's possible to determine the device type (or at least a device with the same coordinates) from `info['dig']`, but it would definitely be a chore.

As far as history, this is not something that is required for writing a valid MFF, we would appreciate the option to include it so that we can log processing steps in the output file.

Reply:
There is also this `info['device_info']` key that could be utilized to store the device type upon reading MFFs and then used to determine the device upon write. I think in any case, at least having the option to provide the device as an argument upon write is a good idea.

Reply:
yes good idea. We could set properly the "device_info" on read from mff file and you remind that there is an info['proc_history'] that we don't use enough. It's just used for maxfilter information AFAIK.
the less stuff we do that is custom the MFF the better. It's more work but I think it's better in the long term.

my 2x